proc/groupby.go

```diff
@@ -343,6 +343,11 @@ func (g *GroupByAggregator) recordsForTable(table map[string]*GroupByRow) []*zso
 	sort.Strings(keys)
 	var recs []*zson.Record
 
+	// We make multiple passes over the table, one for each unique type
+	// vector or row keys, the in each pass we output as a batch all the
+	// records that have this unique type (i.e., the same descriptor).
+	// This allows variations in the type if the join-key to be tracked
```

Comment:
type "of" the join-key

Reply:
fixed (along with the other types in this new paragraph)